js/src/carousel.js

```diff
@@ -351,21 +352,7 @@ class Carousel extends BaseComponent {
 
   _getItemByOrder(order, activeElement) {
     const isNext = order === ORDER_NEXT
-    const isPrev = order === ORDER_PREV
-    const activeIndex = this._getItemIndex(activeElement)
-    const lastItemIndex = this._items.length - 1
-    const isGoingToWrap = (isPrev && activeIndex === 0) || (isNext && activeIndex === lastItemIndex)
-
-    if (isGoingToWrap && !this._config.wrap) {
-      return activeElement
-    }
-
-    const delta = isPrev ? -1 : 1
-    const itemIndex = (activeIndex + delta) % this._items.length
-
-    return itemIndex === -1 ?
-      this._items[this._items.length - 1] :
-      this._items[itemIndex]
+    return pickFromList(this._items, activeElement, isNext, this._config.wrap)
```

Comment:
Maybe we should check for `this._items`? They are initialized by calling `this._getItemIndex()` which is done before (https://github.com/twbs/bootstrap/blob/55006cff8663b981d7db26c10bc9d0d14c56a4b8/js/src/carousel.js#L409). But in case someone changes the order how they're called?

Reply:
I saw it too and you are right. We can check it on a next PR (baby steps to keep it in control)